### PR TITLE
Add an access point to Pi Pico; the microcontroller fully responds to…

### PR DESCRIPTION
**Pull request by ghost**
… the buttons placed on the page

### CODE REVIEW

File: web-app/index.html [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                </style>
+            </head>
+            <body>
+                <nav>
+                    <h1><i>STOPWATCH</i></h1>
+                </nav>
+
+                <div class="timer-container">
+                    <div class="buttons" id="buttons">
+                        <a class="start-button" href="?led=on\" id="start-button">
```
Comment (DoomHammer) [inline web-app/index.html]:
```suggestion
                        <a class="start-button" href="?press=start" id="start-button">
```

File: web-app/index.html [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                            START/STOP
+                        </a>
+                        <a class="reset-button" href="?led=off\" id="reset-button">
```
Comment (DoomHammer) [inline web-app/index.html]:
```suggestion
                        <a class="reset-button" href="?press=reset" id="reset-button">
```

File: web-app/main.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    return html
+
+
+def check_button_click(request):
+    global button300ms, button1000ms
+
+    button300ms, button1000ms = False, False
+
+    led_on = "GET /?led=on" in request.decode("utf-8")
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
    press_start = "GET /?press=start" in request.decode("utf-8")
```

File: web-app/main.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    led_off = "GET /?led=off" in request.decode("utf-8")
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
    press_restart = "GET /?press=restart" in request.decode("utf-8")
```

File: web-app/main.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    if led_on:
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
    if press_start:
```

File: web-app/main.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        print("\nA SHORT click was recorded\n")
+        button300ms = True
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
        short_button = True
```

File: web-app/main.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
+import machine
+import network
+import socket
+import utime
+
+
+# Variables that force further actions such as start/stop/reset time
+button300ms = False
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
short_button = False
```

File: web-app/main.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+button1000ms = False
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
long_button = False
```

File: web-app/main.py [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+def get_template() -> str:
+    with open("index.html", "r") as f:
+        html = f.read()
+
+    return html
+
+
+def check_button_click(request):
+    global button300ms, button1000ms
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
    global short_button, long_button
```

File: web-app/main.py [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+    button300ms, button1000ms = False, False
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
    short_button, long_button = False, False
```

File: web-app/main.py [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+    led_on = "GET /?led=on" in request.decode("utf-8")
+    led_off = "GET /?led=off" in request.decode("utf-8")
+
+    if led_on:
+        print("\nA SHORT click was recorded\n")
+        button300ms = True
+
+    elif led_off:
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
    elif press_restart:
```

File: web-app/main.py [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+        print("\nA LONG click was recorded\n")
+        button1000ms = True
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
        long_button = True
```

File: web-app/main.py [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+        # Turn the LED on/off depending on which button was last pressed
+        if button300ms:
+            utime.sleep_ms(300)
+            led.value(1)
+
+        elif button1000ms:
+            utime.sleep_ms(1000)
+            led.value(0)
+
```
Comment (DoomHammer) [inline web-app/main.py:77]:
```suggestion
        led.value(0)
```

File: web-app/main.py [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+        check_button_click(request)
+
+        # Turn the LED on/off depending on which button was last pressed
+        if button300ms:
+            utime.sleep_ms(300)
+            led.value(1)
+
+        elif button1000ms:
+            utime.sleep_ms(1000)
+            led.value(0)
```
Comment (DoomHammer) [inline web-app/main.py:75]:
```suggestion
            led.value(1)
```

File: web-app/main.py [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+        client, addr = s.accept()
+        request = client.recv(1024)
+
+        print(f"Got a connection from {str(addr)}")
+
+        check_button_click(request)
+
+        # Turn the LED on/off depending on which button was last pressed
+        if button300ms:
+            utime.sleep_ms(300)
```
Comment (DoomHammer) [inline web-app/main.py]: This should go after `led.value()`

File: web-app/main.py [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+            led.value(1)
+
+        elif button1000ms:
+            utime.sleep_ms(1000)
```
Comment (DoomHammer) [inline web-app/main.py]: This should go after `led.value()`

**Review by DoomHammer — CHANGES_REQUESTED** [review]

File: web-app/main.py [hunk 17]
```diff
@@ -61,13 +61,13 @@ def ap_mode(ssid, password):
         check_button_click(request)
 
         # Turn the LED on/off depending on which button was last pressed
-        if button300ms:
+        if short_button:
+            led.value(0)
```
Comment (DoomHammer) [inline web-app/main.py]:
```suggestion
            led.value(1)
```

File: web-app/main.py (hunk 13 above)
Comment (DoomHammer) [reply web-app/main.py:77]: This seems to be still missing?

**Review by DoomHammer — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Now it should be ok. Pin gets high signal for a specific amount of time (onboard LED turns on)  then low signal is captured (LED turns off).

**Review by DoomHammer — APPROVED** [review]

**Review by DoomHammer — APPROVED** [review]

**Pull request merged**